invoice/src/invoice.rs

```diff
@@ -212,17 +212,17 @@ impl Pay2Vout {
     pub(crate) const P2TR: u8 = 5;
 }
 
-impl TryFrom<[u8; 35]> for Pay2Vout {
+impl TryFrom<[u8; 34]> for Pay2Vout {
     type Error = Pay2VoutError;
 
-    fn try_from(data: [u8; 35]) -> Result<Self, Self::Error> {
+    fn try_from(data: [u8; 34]) -> Result<Self, Self::Error> {
         let method =
             CloseMethod::try_from(data[0]).map_err(|e| Pay2VoutError::InvalidMethod(e.1))?;
         let address = match data[1] {
             Self::P2PKH => AddressPayload::Pkh(PubkeyHash::from_slice_unsafe(&data[2..22])),
             Self::P2SH => AddressPayload::Sh(ScriptHash::from_slice_unsafe(&data[2..22])),
             Self::P2WPKH => AddressPayload::Wpkh(WPubkeyHash::from_slice_unsafe(&data[2..22])),
-            Self::P2WSH => AddressPayload::Wsh(WScriptHash::from_slice_unsafe(&data[2..])),
+            Self::P2WSH => AddressPayload::Wsh(WScriptHash::from_slice_unsafe(&data[2..34])),
```

Comment:
The length is 34 now, so this fix is no more required

```suggestion
            Self::P2WSH => AddressPayload::Wsh(WScriptHash::from_slice_unsafe(&data[2..])),
```